python_hll/serialization.py

```diff
@@ -665,7 +665,7 @@ def pack_parameters_byte(cls, register_width, register_count_log2):
         """
         width_bits = (register_width - 1) & cls.REGISTER_WIDTH_MASK
         count_bits = register_count_log2 & cls.LOG2_REGISTER_COUNT_MASK
-        return BitUtil.to_signed_byte(BitUtil.to_signed_byte(BitUtil.left_shift_int(width_bits, cls.LOG2_REGISTER_COUNT_BITS) | count_bits))
+        return BitUtil.left_shift_int(width_bits, cls.LOG2_REGISTER_COUNT_BITS) | count_bits
```

Comment:
The corresponding Java code explicitly casts to a byte, which is why we do it here:

https://github.com/aggregateknowledge/java-hll/blob/1f4126e79a85b79581460c75bf1c1634b8a7402d/src/main/java/net/agkn/hll/serialization/SerializationUtil.java#L209

I agree that we don't have to do it twice though.